qos_controller/serenity_controller.cpp

```diff
@@ -23,18 +22,25 @@ class SerenityControllerProcess :
     public Process<SerenityControllerProcess> {
  public:
   SerenityControllerProcess(
-      const lambda::function<Future<ResourceUsage>()>& _usage)
+      const lambda::function<Future<ResourceUsage>()> _usage)
       : usage(_usage) {}
 
   Future<list<QoSCorrection>> corrections() {
+    return this->usage()
+      .then(defer(self(), &Self::_corrections, lambda::_1));
+  }
+
+  Future<list<QoSCorrection>> _corrections(
+      const Future<ResourceUsage>& _resourceUsage) {
     // TODO(bplotka) Set up the main qos correction pipeline here.
     std::cout << "Serenity QoS Controller pipeline run." << "\n";
-    // For now return empty resources.
+
+    // For now return empty Future.
```

Comment:
I want empty *Future*, since it is not a good idea to return empty list... Slave is registering some callback on this future, so when i fill it with empty list, slave will run its callbacks, check that there is no correction in it, log sth and eventually run *corrections()* (with no delay) and wait for next potential qos corrections... Currently we have empty pipeline so if we are filling this *Future*, it will spam slave - totally unnecessarily:
![image](https://cloud.githubusercontent.com/assets/6950331/8593087/c8ebfe74-2634-11e5-9df6-ccbdf8fc1a9c.png)


IMO even with some real QoS assurance code, QoSController should never return empty list when there are no correction needed. It should fill that *Future* only in case of actual correction. Does it make sense? (: